src/utils/utils.ts

```diff
@@ -45,52 +48,62 @@ export default class Utilities {
-        return splittedSatelitesData.slice(0, splittedSatelitesData.length - 1)
+        return JSON.parse(data)
     }
 
     static countSatelitesFromFile(): number {
-        const data = fs.readFileSync('./ucs-satelite-db.txt', 'utf8')
-        if (!data) return 0
-        return data.split('\n').length - 2
+        return this.getSatelitesFromFile().length as number
     }
 
-    static prePopulateDatabase(satellites: SatelliteInterface[]): Promise<SatelliteInterface[]> {
-        const satellitesCreation: Promise<SatelliteInterface>[] = []
-        satellites.forEach((satellite) => {
-            satellitesCreation.push(SatelliteUtilities.createSatelite(satellite as SatelliteInterface))
-        })
+    static standardizeSatellitesData(satellites: UnprocessedSatellites): SatelliteType[] {
 
-        return Promise.all(satellitesCreation)
-    }
+        const propsToCheck: SatelliteKeys[] = [
+            'longitudeOfGeo',
+            'perigee',
+            'apogee',
+            'inclination',
+            'period',
+            'launchMass',
+            'dryMass',
+            'power',
+            'expectedLifetime',
+            'cospar',
+            'norad'
+        ]
 
-    static checkSatellitesPropsReliability(): {
-        completeSatellites: SatelliteInterface[],
-        incompleteSatellites: SatelliteInterface[]
-    } {
-        const headings = this.satelliteHeaders
-        type SatelliteField = typeof headings[number]
-        type ConstructedSatellite = {
-            [P in SatelliteField]: string | number
+        const addMissingProps = (satellitesList: UnprocessedSatellites) => {
+            return satellitesList.map((item) => {
+                const currentKeys = Object.keys(item)
+                const missigPropertiesList = this.satelliteHeaders.filter((item) => !currentKeys.includes(item))
+                missigPropertiesList.forEach((property) => {
+                    item[property] = ''
+                })
+                return item
+            })
         }
-        const satellitesFromFile = this.getSatelitesFromFile()
 
-        const satellitesCreation = satellitesFromFile.map((satellite) => {
-            const constructedSatellite = headings.reduce((accumulator, heading, index) => {
-                accumulator[heading] = satellite[index] ?? ''
-                return accumulator
-            }, {} as ConstructedSatellite)
-            return constructedSatellite as SatelliteInterface
-        })
+        const replaceCommasWithDots = (unprocessedSatellitesList: UnprocessedSatellites) => {
+            return unprocessedSatellitesList.map((item) => {
+                propsToCheck.forEach((property) => {
+                    if (!item[property]) {
+                        item[property] = 0
+                    } else {
+                        const value = item[property]?.toString() || '0'
```

Comment:
I don't understand why you cannot decide whether you want number here or string.

Reply:
In previous condition you assign zero as a number

Reply:
I just over-engineered it to trick TS, current solution should be fine